--- a/src/backend/flatpak.rs
+++ b/src/backend/flatpak.rs
@@ -47,8 +47,8 @@
 
         let inst = this.installation()?;
         for remote in inst.list_remotes(Cancellable::NONE)? {
-            let source_id = match remote.name() {
-                Some(name) => this.source_id(&name),
+            let (remote_name, source_id) = match remote.name() {
+                Some(name) => (name.to_string(), this.source_id(&name)),
                 None => {
                     log::warn!("remote {:?} missing name", remote);
                     continue;
@@ -80,7 +80,7 @@
 
             if update_appstream {
                 log::info!("updating appstream data for remote {:?}", remote);
-                match inst.update_appstream_sync(&source_id, None, Cancellable::NONE) {
+                match inst.update_appstream_sync(&remote_name, None, Cancellable::NONE) {
                     Ok(()) => {}
                     Err(err) => {
                         log::warn!(
